pycheribuild/projects/project.py

```diff
@@ -2622,7 +2623,10 @@ def configure(self, **kwargs):
             # CMAKE_CROSSCOMPILING will be set when we change CMAKE_SYSTEM_NAME:
             # This means we may not need the toolchain file at all
             # https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html
-            system_name = "Generic" if self.target_info.is_baremetal else "FreeBSD"
```

Comment:
This should probably be an abstract property in TargetInfo (similar to `cmake_processor_id`, but that change can be made later.